lib/Parse/ParseDecl.cpp

```diff
@@ -3383,7 +3435,29 @@ bool Parser::isStartOfDecl() {
   // Otherwise, do a recursive parse.
   Parser::BacktrackingScope Backtrack(*this);
   consumeToken(tok::identifier);
-  return isStartOfDecl();
+  return isStartOfSwiftDecl();
+}
+
+bool Parser::isStartOfSILDecl() {
+  switch (Tok.getKind()) {
+  case tok::kw_sil:
+  case tok::kw_sil_stage:
+  case tok::kw_sil_property:
+  case tok::kw_sil_vtable:
+  case tok::kw_sil_global:
+  case tok::kw_sil_witness_table:
+  case tok::kw_sil_default_witness_table:
+  case tok::kw_sil_coverage_map:
+  case tok::kw_sil_scope:
+    return true;
```

Comment:
I believe that SIL keywords are escaped if developers use them in their swift files (e.g. `struct sil {}` becomes ``struct `sil` {}`` in SIL), but for *extra* safety, how about checking `isAtStartOfLine()` check here?

Reply:
@rintaro Sure thing! I've gone ahead and made that change.

Thanks for the review!